### ISSUE TITLE
Possible Bug: History consists of only `loss` and `val_loss`

### ISSUE THREAD

**Issue by soumik12345** [keras-team/keras-core#485]
I am currently attempting to convert the example [Zero-DCE for low-light image enhancement](https://keras.io/examples/vision/zero_dce/) to `keras_core`. The example demonstrates overriding `train_step()` and `test_step()`, both of which return a dictionary of metrics like [this](https://github.com/keras-team/keras-io/blob/master/examples/vision/zero_dce.py#L353-L359). However, when I check the keys present in `history.history`(history is of type `keras_core.callbacks.history.History`), I find only `'loss'` and `'val_loss'`.

Colab Notebook for reproducing the issue: https://colab.research.google.com/drive/1GuumdRuh9wy_Eo_zzORFfG2iX8gGG58a?usp=sharing

**Comment by fchollet** [comment]
The output of `train_step` is reported in the progress bar, but the history logs at the epoch level don't come from there. They come from the `self.metrics` property. So what you need to do is:

- Create `Mean` trackers for the different metrics to track
- Call `update_state()` on them in `train_step`

QQ: did this work in tf.keras? If so, how?

**Comment by soumik12345** [comment]
Hi @fchollet 
Tried implementing your suggestion. It didn't work for me ([reference colab](https://colab.research.google.com/drive/1GuumdRuh9wy_Eo_zzORFfG2iX8gGG58a?usp=sharing)).

> QQ: did this work in tf.keras? If so, how?

It seemed to work in `tf.keras`. It was pulled from `history.history` in the [original example](https://keras.io/examples/vision/zero_dce/).

**Comment by fchollet** [comment]
Your code works fine, actually. You're querying the wrong key (you named it `total_loss_tracker` not `total_loss`). I see the following keys:

```
dict_keys(['color_constancy_loss', 'exposure_loss', 'illumination_smoothness_loss', 'spatial_constancy_loss', 'total_loss_tracker', 'val_color_constancy_loss', 'val_exposure_loss', 'val_illumination_smoothness_loss', 'val_spatial_constancy_loss', 'val_total_loss_tracker'])
```

**Comment by fchollet** [comment]
> It seemed to work in tf.keras

Can you look into how that worked?

**Comment by soumik12345** [comment]
> Your code works fine, actually. You're querying the wrong key (you named it `total_loss_tracker` not `total_loss`). I see the following keys:
> 
> ```
> dict_keys(['color_constancy_loss', 'exposure_loss', 'illumination_smoothness_loss', 'spatial_constancy_loss', 'total_loss_tracker', 'val_color_constancy_loss', 'val_exposure_loss', 'val_illumination_smoothness_loss', 'val_spatial_constancy_loss', 'val_total_loss_tracker'])
> ```

Thanks for point out! Updated https://github.com/keras-team/keras-core/pull/486 accordingly.
